### PR TITLE
Added fatal checker configuration option in SQSMessagePollerOptions 

### PR DESCRIPTION
**Pull request by madhub**
*Issue #, if available:*
#56 

*Description of changes:*

Currently [SQSMessagePoller](https://github.com/awslabs/aws-dotnet-messaging/blob/d826b7934268942a5c81b19e7aeacf3ba5794d2d/src/AWS.Messaging/SQS/SQSMessagePoller.cs#L127) only checks "InvalidAddress" and "AccessDenied" decided whether to continue polling or rethrow the exception. This is not sufficient , example "QueueDoesNotExistException" exception also should stop the polling.

Better design option would be provide extensibility in SQSMessagePollerOptions that can take user supplied function to decide when top stop polling

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Messaging/Configuration/SQSMessagePollerConfiguration.cs [hunk 1]
```diff
@@ -113,4 +119,22 @@ internal MessageManagerConfiguration ToMessageManagerConfiguration()
...
         };
     }
+
+    /// <summary>
+    /// <see cref="AmazonSQSException"/> error codes that should be treated as fatal and stop the poller
+    /// </summary>
+    private static readonly HashSet<string> _fatalSQSErrorCodes = new HashSet<string>
+    {
+        "InvalidAddress",   // Returned for an invalid queue URL
+        "AccessDenied"      // Returned with insufficient IAM permissions to read from the configured queue
```
Comment (ashovlin) [inline src/AWS.Messaging/Configuration/SQSMessagePollerConfiguration.cs:129]: Feel free to add `QueueDoesNotExistException` in here. Sorry for not replying to #56 sooner, but I agree it should be in the default error list.

File: src/AWS.Messaging/Configuration/SQSMessagePollerConfiguration.cs [hunk 2]
```diff
@@ -87,6 +88,11 @@ internal class SQSMessagePollerConfiguration : IMessagePollerConfiguration
     /// </remarks>
     public int WaitTimeSeconds { get; init; } = DEFAULT_WAIT_TIME_SECONDS;
 
+    /// <summary>
+    /// Function that that indicated whether SQS poller should stop polling or continue;
+    /// </summary>
+    public Func<AmazonSQSException, bool> FatalErrorChecker { get; set; } = IsSQSExceptionFatal;
```
Comment (ashovlin) [inline src/AWS.Messaging/Configuration/SQSMessagePollerConfiguration.cs:94]: Suggestion to clean up documentation and naming:
```suggestion
    /// Function that indicates whether an <see cref="AmazonSQSException"/> should stop the poller or continue.
    /// </summary>
    public Func<AmazonSQSException, bool> IsSQSExceptionFatal { get; set; } = IsSQSExceptionFatal;
```

**Comment by ashovlin** [comment]
One more spot to add the new configuration, can you copy it from the `SQSMessagePollerOptions` to the `SQSMessagePollerConfiguration` here:
https://github.com/awslabs/aws-dotnet-messaging/blob/d826b7934268942a5c81b19e7aeacf3ba5794d2d/src/AWS.Messaging/Configuration/MessageBusBuilder.cs#L119-L126

It's a little awkward, but we arrived here because the queueURL is configured outside of the `Action<SQSMessagePollerOptions>` since it's the primary/required input.

**Review by ashovlin — CHANGES_REQUESTED** [review]
One more spot to add the new configuration, can you copy it from the `SQSMessagePollerOptions` to the `SQSMessagePollerConfiguration` here:
https://github.com/awslabs/aws-dotnet-messaging/blob/d826b7934268942a5c81b19e7aeacf3ba5794d2d/src/AWS.Messaging/Configuration/MessageBusBuilder.cs#L119-L126

It's a little awkward, but we arrived here because the queueURL is configured outside of the `Action<SQSMessagePollerOptions>` since it's the primary/required input.

**Review by ashovlin — APPROVED** [review]

**Review by normj — APPROVED** [review]
Thanks for the PR!

**Pull request merged**